Updated is_ref to make unmarshal_response ~30% faster

ClareCat opened the pull request:
Updated is_ref to reduce the number of times is_dict_like gets called. Instance checking is a huge source of slowdown because of the number of times isinstance gets called. Did some comparisons between the benchmarks seen in https://travis-ci.org/Yelp/bravado-core/jobs/264892454 and https://travis-ci.org/Yelp/bravado-core/jobs/264894131

| Py27 - Validated | Old Mean (ms) | New Mean (ms) | Change (% faster) |
|------------------|---------------|---------------|-------------------|
| 1000 small obj   | 201.2373      | 152.8966      | 24.02%            |
| 5000 small obj   | 995.7440      | 783.1190      | 21.35%            |
| 1000 large obj   | 716.8537      | 527.9353      | 26.35%            |
| 5000 large obj   | 3,748.9801    | 2,718.8094    | 27.48%            |

| Py36 - Validated | Old Mean (ms) | New Mean (ms) | Change (% faster) |
|------------------|---------------|---------------|-------------------|
| 1000 small obj   | 191.4858      | 138.5805      | 27.63%            |
| 5000 small obj   | 959.6830      | 704.6253      | 26.58%            |
| 1000 large obj   | 720.0888      | 493.1808      | 31.51%            |
| 5000 large obj   | 3,592.0855    | 2,496.4199    | 30.5%             |

| Py27 - Not Validated | Old Mean (ms) | New Mean (ms) | Change (% faster) |
|----------------------|---------------|---------------|-------------------|
| 1000 small obj       | 110.8196      | 82.8784       | 25.21%            |
| 5000 small obj       | 566.4803      | 474.1994      | 16.29%            |
| 1000 large obj       | 376.0710      | 248.5105      | 33.92%            |
| 5000 large obj       | 1,845.8902    | 1,320.3167    | 28.47%            |

| Py36 - Not Validated | Old Mean (ms) | New Mean (ms) | Change (% faster) |
|----------------------|---------------|---------------|-------------------|
| 1000 small obj       | 117.8748      | 80.1987       | 31.96%            |
| 5000 small obj       | 581.2813      | 408.2662      | 29.76%            |
| 1000 large obj       | 386.8550      | 251.3254      | 35.03%            |
| 5000 large obj       | 1,945.9027    | 1,287.6521    | 33.83%            |

laucia commented on bravado_core/schema.py:
```diff
@@ -47,7 +47,10 @@ def is_prop_nullable(swagger_spec, schema_object_spec):
 
 def is_ref(spec):
-    return is_dict_like(spec) and '$ref' in spec
+    try:
```
this is not the most obviously pythonic form for the function, can you add a comment here as to why the function is written that way ?
So we don't change it back accidentally :)

ClareCat commented:
The benchmarking tests were making testing much slower (especially during dev) so I adjusted the make target for 'test' to add args that will ignore the benchmarking tests and added a 'benchmark' target. Travis will still run all of the tests because it just runs tox without any of the ignore posargs.

Noticed during this that the way the tox environments were setup it was impossible to pass in any posargs for pytest that wouldn't crash in pre-commit so opted to remove posargs from the pre-commit testenv.
